lib/obj.js

```diff
@@ -656,6 +656,17 @@ function saveMetadata(req, res, next) {
                     res.header('Access-Control-Allow-Origin',
                                req.headers['origin']);
                 }
+                /*
+                 * The following headers will have already been set to the
+                 * previous metadata values if they existed (for example, if
+                 * the object already existed and is being overwritten).
+                 * Because of this, they should be removed before being set to
+                 * the new values to ensure multiple headers are not sent for
+                 * the following keys.
+                 */
+                res.removeHeader('Etag');
+                res.removeHeader('Last-Modified');
+                res.removeHeader('Computed-MD5');
```

Comment:
I don't think Computed-MD5 is set anywhere else, FWIW.  Doesn't harm to keep this here.